### PR TITLE
Improvement to IDE-friendliness when 'wrapper_class' option is used

### PR DESCRIPTION
**Pull request by Xymanek**
Sets the `class` property in service definition to assist IDEs in knowing what class the service really is. Since connection services use a factory, the `class` option doesn't affect anything. I've tested the logic with following "hack" and it works without any problems:

```php
class DataConnectionCompilerPass implements CompilerPassInterface
{
    public function process (ContainerBuilder $container)
    {
        $service = $container->findDefinition('doctrine.dbal.data_connection');
        $service->setClass(DataConnectionWrapper::class);
    }
}
```

P.S. Why so many `private`s in `\Doctrine\DBAL\Connection`? Makes extending it so hard....

### CODE REVIEW

File: Tests/DependencyInjection/DoctrineExtensionTest.php [hunk 1]
```diff
@@ -196,6 +197,28 @@ public function testDbalLoad()
...
+                ))),
+                array(),
+                array('dbal' => array('default_connection' => 'foo')),
+                array()
+            ),
+            $container
+        );
+
+        $this->assertEquals(TestWrapperClass::class, $container->getDefinition('doctrine.dbal.default_connection')->getClass());
+        $this->assertEquals(null, $container->getDefinition('doctrine.dbal.second_connection')->getClass());
```
Comment (stloyd) [inline Tests/DependencyInjection/DoctrineExtensionTest.php]: `assertNull()`

Comment (Xymanek) [inline Tests/DependencyInjection/DoctrineExtensionTest.php]: Ooops

Comment (Xymanek) [inline Tests/DependencyInjection/DoctrineExtensionTest.php]: Anything else? Don't want to pollute with commits

**Comment by Ocramius** [comment]
> P.S. Why so many privates in `\Doctrine\DBAL\Connection`? Makes extending it so hard....

It's not meant to be extended.

**Comment by Xymanek** [comment]
Anything else needs to be done?

File: DependencyInjection/DoctrineExtension.php [hunk 2]
```diff
@@ -221,6 +221,11 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
             ));
             $container->setDefinition(sprintf('doctrine.dbal.%s_shard_manager', $name), $shardManagerDefinition);
         }
+
+        // Set class in case "wrapper_class" option was used to assist IDEs
+        if (isset($options['wrapperClass'])) {
+            $def->setClass($options['wrapperClass']);
+        }
```
Comment (weaverryan) [inline DependencyInjection/DoctrineExtension.php]: I would move this up to after line 211 - so right after `$def = `

**Comment by weaverryan** [comment]
+1 The `wrapperClass` option is ultimately used in the DBAL to create a class of that specific instance. So, yea, technically, the definition's should match the actual class that will be created.

**Comment by Xymanek** [comment]
@weaverryan Fixed and fixed

**Comment by Ocramius** [comment]
@Xymanek can you please rebase instead of merging `master`? Commit Xymanek/DoctrineBundle@124573478feb51e4908e72077f1c06982d80ff15 should go away

**Comment by Xymanek** [comment]
@Ocramius can you please give me a hint on how to do that? I kinda know what you mean but not sure how to do it cross-repository...

**Comment by Ocramius** [comment]
@Xymanek I'll do that now while merging :-)

If you never did it before, consider reading https://www.atlassian.com/git/tutorials/rewriting-history

**Comment by Ocramius** [comment]
Manually rebased and merged via doctrine/DoctrineBundle@1818d3d60bff83bdeea79cbb852eed104217998f

Thanks @Xymanek!

**Pull request closed without merging**